kolibri/core/assets/src/composables/useNav.js

```diff
@@ -0,0 +1,87 @@
+import useKResponsiveWindow from 'kolibri-design-system/lib/composables/useKResponsiveWindow';
+import { KolibriIcons } from 'kolibri-design-system/lib/KIcon/iconDefinitions';
+import { get } from '@vueuse/core';
+import { UserKinds, NavComponentSections } from 'kolibri.coreVue.vuex.constants';
+import logger from 'kolibri.lib.logging';
+import { computed } from 'kolibri.lib.vueCompositionApi';
+
+const logging = logger.getLogger(__filename);
+
+export const navComponents = [];
+
+function checkDeclared(property) {
+  return typeof property !== 'undefined' && property !== null;
+}
+
+function validateUrl(url) {
+  return checkDeclared(url) && typeof url === 'string';
+}
+
+function validateIcon(icon) {
+  return checkDeclared(icon) && typeof icon === 'string' && Boolean(KolibriIcons[icon]);
+}
+
+function validateRole(role) {
+  // Optional, must be one of the defined UserKinds
+  return !checkDeclared(role) || Object.values(UserKinds).includes(role);
+}
+
+function validateSection(section) {
+  // Optional, must be one of the defined NavComponentSections
+  return !checkDeclared(section) || Object.values(NavComponentSections).includes(section);
+}
+
+function validateRoutes(routes) {
+  // Required, must be an array of objects
```

Comment:
So, the url is the plugin "home" URL, and these are the sub-routes. I think this is not always required. because of things like the login/out side nav entry and profile. and the spec suggests that just url and icon are required for it to be registered

Reply:
Yes - it appears the comment is out of kilter with what the validation is actually doing, where it definitely does allow these not to be defined.